app.js

```diff
@@ -1,14 +1,52 @@
 const http = require('http');
+const fs = require('fs');
 
 const hostname = '127.0.0.1';
 const port = 3000;
 
-const server = http.createServer((req, res) => {
-  res.statusCode = 200;
-  res.setHeader('Content-Type', 'text/plain');
-  res.end('Hello World');
+
+// using a read stream. Don't fully understand this
+const htmlPipeFileServer = http.createServer((req, res) => {
```

Comment:
I don't fully understand the readstream().pipe() workflow.

Maybe pipe takes a callback? That would make sense. 

Bung all the parsed data into an object and then call `res` with this object.